### DISCUSSION TITLE
How hard would it be for me to replace MongoDB with another database in my implementation?

### DISCUSSION THREAD

**Discussion by drumrz** [danny-avila/LibreChat#1707] [Help Wanted]
Obviously anything is possible, but how embedded is Mongo in the app? Would it be just a few files to change to change the database to something like ElasticSearch, or a total redesign?

**Comment by danny-avila** [comment]
I answered this [on discord](https://discord.com/channels/1086345563026489514/1105648303909126144/1201650374214422528) the other day. In short, it's not just a few files needing changed, but not a complete redesign (but leaning more to this in terms of volume and knowledge of where database use is applied).

Replacing mongoDB is a long path forward, I estimated it would take me a month to do this myself, and that's with me knowing the codebase really well. I've been writing any use of database controllers with as little to no mongo-specific features in case I ever find the time/reason to do this one day.

You have to be pretty familiar with the codebase, and then design all the schemas in a way that work.

I would like to work on this one day but other things seem more pressing at the moment, so I keep that mentality of keep things as non-mongo-specific as possible.

**Reply by drumrz** [reply]
Thank you for this.
